.github/scripts/mr_publish_results.py

```diff
@@ -0,0 +1,66 @@
+# Send event to Segment with a summary of test
+import analytics
+import datetime
+import json
+import os
+
+analytics.write_key = os.environ['SEGMENT_TOKEN']
+
+task_mapping = {
+    "intent_report.json": "Intent Classification",
+    "CRFEntityExtractor_report.json": "Entity Prediction",
+    "DIETClassifier_report.json": "Entity Prediction",
+    "response_selection_report.json": "Response Selection"
+}
+
+def send_to_segment(context):
+    jobID = os.environ['GITHUB_RUN_ID']
+
+    analytics.identify(jobID, {
+        'name': 'model-regression-tests',
+        'created_at': datetime.datetime.now()
+    })
+
+    analytics.track(jobID, 'results', {
+        'dataset': os.environ['DATASET_NAME'],
+        'workflow': os.environ['GITHUB_WORKFLOW'],
+        'config': os.environ['CONFIG'],
+        'pr_url': os.environ['PR_URL'],
+        'accelerator_type': os.environ['ACCELERATOR_TYPE'],
+        'test_run_time': os.environ['TEST_RUN_TIME'],
+        'train_run_time': os.environ['TRAIN_RUN_TIME'],
+        'total_run_time': os.environ['TOTAL_RUN_TIME'],
+        'github_run_id': os.environ['GITHUB_RUN_ID'],
+        'github_sha': os.environ['GITHUB_SHA'],
+        'event': os.environ['GITHUB_EVENT_NAME'],
+        **context
+    })
+
+def read_results(file):
+    with open(file) as json_file:
+        data = json.load(json_file)
```

Comment:
optional: should we put everything into one script since there is some duplicate logic / code?

Reply:
We don't always use both scripts, for the workflow which runs on schedule event we use only one of them. For now, I would keep them separate.